mypy/plugin.py

```diff
@@ -379,3 +421,332 @@ def int_pow_callback(ctx: MethodContext) -> Type:
+    """
+    # attrs is a package that lets you define classes without writing dull boilerplate code.
+    #
+    # At a quick glance, the decorator searches the class body for assignments of `attr.ib`s (or
+    # annotated variables if auto_attribs=True), then depending on how the decorator is called,
+    # it will add an __init__ or all the __cmp__ methods.  For frozen=True it will turn the attrs
+    # into properties.
+    #
+    # See http://www.attrs.org/en/stable/how-does-it-work.html for information on how attrs works.
+
+    def called_function(expr: Expression) -> Optional[str]:
+        """Return the full name of the function being called by the expr, or None."""
+        if isinstance(expr, CallExpr) and isinstance(expr.callee, RefExpr):
+            return expr.callee.fullname
+        return None
+
+    def get_callable_type(call: CallExpr) -> Optional[CallableType]:
+        """Get the CallableType that's attached to a CallExpr."""
+        callable_type = None
+        if (isinstance(call, CallExpr)
+                and isinstance(call.callee, RefExpr)
+                and isinstance(call.callee.node, Var)
+                and call.callee.node.type):
+            callee_type = call.callee.node.type
+            if isinstance(callee_type, Overloaded):
+                # We take the last overload.
+                callable_type = callee_type.items()[-1]
+            elif isinstance(callee_type, CallableType):
+                callable_type = callee_type
+        return callable_type
+
+    def get_argument(call: CallExpr, name: str) -> Optional[Expression]:
+        """Return the expression for the specific argument."""
+        callee = get_callable_type(call)
+        if not callee:
+            return None
+
+        argument = callee.argument_by_name(name)
+        if not argument:
+            return None
+
+        if not argument.name and not argument.pos:
+            return None
+
+        for i, (attr_name, attr_value) in enumerate(zip(call.arg_names, call.args)):
+            if argument.pos is not None and not attr_name and i == argument.pos:
+                return attr_value
+            if attr_name == argument.name:
+                return attr_value
+        return None
+
+    def get_bool_argument(expr: Expression, name: str, default: bool) -> bool:
+        """Return the value of an argument name in the give Expression.
+
+        If it's a CallExpr and the argument is one of the args then return it.
+        Otherwise return the default value for the argument.
+        """
+        if isinstance(expr, CallExpr):
+            attr_value = get_argument(expr, name)
+            if attr_value:
+                ret = ctx.api.parse_bool(attr_value)
+                if ret is None:
+                    ctx.api.fail('"{}" argument must be True or False.'.format(name), expr)
+                    return default
+                return ret
+        return default
```

Comment:
Add a comment that a decorator can be only ``CallExpr`` or ``RefExpr``, therefore we can get here with a plain ``attr.s``. Also I don't it is safe to apply the same logic to ``attr.ib`` as you do below for ``init``.

Reply:
This is still not fixed.